src/ray/raylet/scheduling/cluster_task_manager_interface.h

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "ray/raylet/worker.h"
+#include "ray/rpc/server_call.h"
+#include "src/ray/protobuf/node_manager.pb.h"
+
+namespace ray {
+namespace raylet {
+class ClusterTaskManagerInterface {
```

Comment:
There are now three different methods (FreeLocalTaskResources, HandleTaskFinished, and ReturnWorkerResources) that all return resources that were used by a worker. I found this pretty confusing because it's hard to tell which method should be called when from the NodeManager.

Can we split out the common code into one method called "Free/ReleaseWorkerResources" that gets called by the NodeManager? If necessary, you can move any other logic into a separate interface method (with a note that it should get deleted with the legacy scheduler).

Reply:
Good suggestion, I will do it.